example/screens/POIExample.tsx

```diff
@@ -0,0 +1,119 @@
+
+import * as Maps from 'expo-maps';
+import SwitchContainer from '../components/SwitchContainer';
+import ProviderContext from '../context/ProviderContext';
+
+export default function POIExample() {
+  const provider = useContext(ProviderContext);
+  const ref = useRef<Maps.ExpoMap>(null);
+  const [enablePOISearching, setEnablePOISearching] = useState<boolean>(false);
+  const [enablePOIDisplay, setEnablePOIDisplay] = useState<boolean>(false);
+  const [enablePOIFilter, setEnablePOIFilter] = useState<boolean>(false);
+  const [poiType, setPoiType] = useState<[] | [Maps.POICategoryType]>([]);
+  const [enableQueryCompletions, setEnableQueryCompletions] =
+    useState<boolean>(false);
+  const [text, onChangeText] = useState<string>('');
+  const [placeToSearch, setPlaceToSearch] = useState<string>('');
+  const [enablePlaceSearch, setEnablePlaceSearch] = useState<boolean>(false);
+
+  const appleMapsSearchRequest = 'Centrum Pompidou; Roue 1234';
+  const googleMapsSearchRequest =
+    'Centrum Pompidou; placeId:ChIJoyC4CRxu5kcRRTPcWX5srLc';
+
+  return (
+    <View style={styles.mapContainer}>
+      <Maps.ExpoMap
+        style={{ flex: 1, width: '100%' }}
+        provider={provider}
+        enablePOISearching={enablePOISearching}
+        enablePOIDisplay={enablePOIDisplay}
+        enablePOIFilter={poiType}
+        createPOISearchRequest={placeToSearch}
+        ref={ref}
+        initialCameraPosition={{
+          latitude: 48.85,
+          longitude: 2.34,
+          zoom: 13,
+          animate: true,
+        }}
+      />
+      <View style={styles.switchContainer}>
+        {provider == 'apple' && (
+          <SwitchContainer
+            title="Enable POI search"
+            value={enablePOISearching}
+            onValueChange={() => setEnablePOISearching(!enablePOISearching)}
+          />
+        )}
+        <SwitchContainer
+          title="Display POI"
+          value={enablePOIDisplay}
+          onValueChange={() => setEnablePOIDisplay(!enablePOIDisplay)}
+        />
+        {provider == 'apple' && (
+          <SwitchContainer
+            title="Enable POI cafe filter"
+            value={enablePOIFilter}
+            onValueChange={() => {
+              setEnablePOIFilter(!enablePOIFilter);
+              if (enablePOIFilter) setPoiType([]);
+              else setPoiType(['cafe']);
+            }}
+          />
+        )}
+        <SwitchContainer
+          title="Fetch query completions (display in console)"
+          value={enableQueryCompletions}
+          onValueChange={() =>
+            setEnableQueryCompletions(!enableQueryCompletions)
+          }
+        />
+        <TextInput
+          style={styles.textInput}
+          editable={enableQueryCompletions}
+          onChangeText={async (text) => {
+            onChangeText(text);
+            await ref.current?.getSearchCompletions(text);
+          }}
+          placeholder={'Search query'}
+          value={text}
+        />
```

Comment:
We can work on styling of this component and put it on top of the example view (above the map), so the keyboard doesn't cover this input when typing. I mean it doesn't necessarily need to be done as a part of this PR, but it would be good to have it improved some time soon. Also I would say that we should somehow indicate that this component is inactive, when search is disabled, or hide it from UI. I can take care about it in separate PR to not block this one. 

We can even do another improvement and create our custom component to display the search results so they are not only visible in the console.

What do you think?

Reply:
Good idea, I created an Issue with suggested changes on this topic https://github.com/software-mansion-labs/expo-maps/issues/81

Reply:
Thanks!